astropy/io/fits/fitstime.py

```diff
@@ -245,18 +356,52 @@ def _convert_time_column(col, column_info):
     """
 
     # The code might fail while attempting to read FITS files not written by astropy.
+    try:
+        # ISO-8601 is the only string representation of time in FITS
+        if col.info.dtype.kind in ['S', 'U']:
+            # [+/-C]CCYY-MM-DD[Thh:mm:ss[.s...]] where the number of characters
+            # from index 20 to the end of string represents the precision
```

Comment:
I think the `20` is only correct if the time does not use the `+/-C`; that said, this is only a guess, and we don't really want to start to inspect all the strings, so I think it is OK (in fact, if we were to inspect, it should probably inside the `TimeFits` format).